### PR TITLE
add elixir 1.7 support

### PR DESCRIPTION
**Pull request by ayrat555**
Changes:
- fix compilation warnings for Elixir 1.7
- fix `Stream.cycle` for empty enum (elixir 1.5 change - https://github.com/elixir-lang/elixir/blob/v1.5/CHANGELOG.md#elixir-6)

### CODE REVIEW

File: lib/progress_bar/bar_formatter.ex [hunk 1]
```diff
@@ -11,41 +11,43 @@ defmodule ProgressBar.BarFormatter do
...
     ]
   end
 
   defp full_bar_width(format, suffix) do
-    other_text = chardata(format, "", suffix) |> IO.chardata_to_string
+    other_text = chardata(format, "", suffix) |> IO.chardata_to_string()
     ProgressBar.FullBarWidth.determine(format[:width], other_text)
   end
 
+  defp repeat("", _), do: ""
```
Comment (ayrat555) [inline lib/progress_bar/bar_formatter.ex:44]: fixes Stream.cycle for empty enum

**Comment by ayrat555** [comment]
@henrik can you please review this PR?

**Comment by henrik** [comment]
Hi @ayrat555! Thanks so much for the PR. Hope to get around to it within the next few days.

**Comment by ayrat555** [comment]
@henrik ping

**Comment by henrik** [comment]
Sorry about that, merging it now! Life gets busy.

Are all the formatting changes required to avoid warnings? Or are they the result of the Elixir code formatter? Some of them run counter to my preferences (I see reasons to keep trailing commas on the last line of an array, and moving an end-of-line comment above it without extra blank lines blurs the comment scope) but I'll take them if that's what Elixir demands :D

**Comment by henrik** [comment]
Released the package as 1.7.0.

**Comment by ayrat555** [comment]
thanks

**Pull request merged**